mlflow/genai/optimize/types.py

```diff
@@ -118,3 +118,52 @@ class OptimizerOutput:
     optimizer_name: str
     final_eval_score: float | None = None
     initial_eval_score: float | None = None
+
+
+@experimental(version="3.5.0")
+@dataclass
+class EvaluationResultRecord:
+    """
+    An output type of `eval_fn` in the
+    :py:func:`mlflow.genai.optimize.BasePromptAdapter.optimize()` API.
+
+    Args:
+        inputs: The inputs of the evaluation.
+        outputs: The outputs of the evaluation.
+        score: The score of the evaluation result.
+        trace: The trace of the evaluation execution.
+    """
+
+    inputs: dict[str, Any]
+    outputs: Any
+    score: float
+    trace: Any
+
+
+@experimental(version="3.5.0")
+@dataclass
+class PromptAdapterOutput:
+    """
+    An output of the :py:func:`mlflow.genai.optimize.BasePromptAdapter.optimize()` API.
+
+    Args:
+        optimized_prompts: The optimized prompts as
+            a dict (prompt template name -> prompt template).
```

Comment:
An example could be helpful. Also do we need this public class? it looks very similar to `PromptAdaptationResult` from the name, and `PromptAdapterOutput` just holds a dict.